src/main/java/com/pollistics/models/Poll.java

```diff
@@ -16,22 +16,46 @@
 	private String name;
 	private HashMap<String,Integer> options;
 	private User user;
+	private String slug;
+
+	// todo: orden these constructors with `this(...arg)`
 
 	public Poll(String name, HashMap<String,Integer> options) {
 		this.name = name;
 		this.options = options;
+		this.slug = createSlug();
 	}
 
-	public Poll(String name, HashMap<String,Integer> options, User user) {
+	private String createSlug() {
+		// random three-word combo
+		return "meme-meme-meme";
```

Comment:
We should create a mongodb unique index on the slug field in the poll collection to keep it performant. A possible solution would be to generate a random 3 word string and if mongo rejects it (because it isn't unique), generate another one. Not sure if this is the best approach.